### PR TITLE
Fix homing child weapons in niche circumstances

### PR DESCRIPTION
**Pull request by EatThePath**
This line, in code that runs if a seeker's range to target ever increases noticeably, removes the target_num but doesn't touch the values that are primarily used for homing. The only effect it has that I've been able to find is to break the inherit parent target weapon flag, which checks the target number.

### CODE REVIEW

**Review by Baezon — APPROVED** [review]

**Review by Goober5000 — CHANGES_REQUESTED** [review]
One thing that *will* change is that it will cause the "If our target is still valid, then update some info." block to be run every frame, since it checks `target_num`.  Will this have any effect?

**Comment by Baezon** [comment]
After some investigation, it appears that both `lead_scale` and `nearest_dist` are used by literally nothing. They are initialized elsewhere and updated by this block, and that's it. >.>

Given that this whole block might just be deleted. This is clearly not a 'lost your target' situation so it really shouldn't be setting `target_num` to -1.

**Comment by EatThePath** [comment]
> 
> 
> After some investigation, it appears that both `lead_scale` and `nearest_dist` are used by literally nothing. They are initialized elsewhere and updated by this block, and that's it. >.>
> 
> Given that this whole block might just be deleted. This is clearly not a 'lost your target' situation so it really shouldn't be setting `target_num` to -1.

I can expand the pr to just clean that all out then, in that case.

**Comment by Baezon** [comment]
Make sure to give it another round of testing as well, to be sure.

**Comment by EatThePath** [comment]
@asarium sorry about that, sloppy keyboarding and sloppy git-fu accidentally made a spurious white-space change to a scripting file, since force-pushed out of the timeline.

**Comment by EatThePath** [comment]
> 
> 
> One thing that _will_ change is that it will cause the "If our target is still valid, then update some info." block to be run every frame, since it checks `target_num`. Will this have any effect?

As Asteroth indicated, the data being updated is data only ever used in this section of the code anyway. As far as I can tell everything in the if block indicated by that comment might as well be a noop as far as the rest of the simulation is concerned, except the line I was deleting to begin with. With my last push, it's all torn out now.

**Review by Goober5000 — APPROVED** [review]
Ok.

**Comment by Goober5000** [comment]
Let's not merge this until after 21.4 is out though.

**Pull request merged**
